src/lib/Key.js

```diff
@@ -0,0 +1,250 @@
+/**
+ * Usage:
+ * <script src="lib/key.js"></script>
+ *
+ * const key = new Key(keypair, type);
+ * const key = await Key.loadEncrypted('encrypted keypair as hex', passphrase);
+ *
+ * const tx = key.createTransaction(recipient, value, fee, validityStartHeight);
+ * const tx = key.createTransactionWithMessage(recipient, value, fee, validityStartHeight, message);
+ * const tx = key.createVestingPayoutTransaction(sender, value, fee, validityStartHeight, [message]);
+ * const tx = key.createExtendedTransaction(sender, [senderType], recipient, [recipientType], value, fee, validityStartHeight, [extraData], [isContractCreation]);
+ *
+ * const msg = key.signMessage('message');
+ */
+class Key {
+    /**
+     * @param {Uint8Array|string} buf - Keypair, as byte array or HEX string
+     * @returns {Key}
+     */
+    static loadPlain(buf) {
+        if (typeof buf === 'string') buf = /** @type {Uint8Array} */ (Nimiq.BufferUtils.fromHex(buf));
+        if (!buf || buf.byteLength === 0) {
+            throw new Error('Invalid Key seed');
+        }
+        return new Key(Nimiq.KeyPair.unserialize(new Nimiq.SerialBuffer(buf)));
+    }
+
+    /**
+     * @param {Uint8Array|string} buf - Encrypted keypair, as byte array or HEX string
+     * @param {Uint8Array|string} passphrase - Passphrase, as byte array or ASCII string
+     * @returns {Promise.<Key>}
+     */
+    static async loadEncrypted(buf, passphrase) {
+        if (typeof buf === 'string') buf = /** @type {Uint8Array} */ (Nimiq.BufferUtils.fromHex(buf));
+        if (typeof passphrase === 'string') passphrase = /** @type {Uint8Array} */ (Nimiq.BufferUtils.fromAscii(passphrase));
+        return new Key(await Nimiq.KeyPair.fromEncrypted(new Nimiq.SerialBuffer(buf), passphrase));
+    }
+
+
+    /**
+     * @param {string} friendlyAddress
+     * @returns {Nimiq.Address}
+     */
+    static getUnfriendlyAddress(friendlyAddress) {
+        return Nimiq.Address.fromUserFriendlyAddress(friendlyAddress);
+    }
+
+    /**
+     * Create a new Key object.
+     * @param {Nimiq.KeyPair} keyPair - Keypair for this key
+     * @param {1|2} [type] - Low or high security (passphrase or pin encoded, respectively)
+     */
+    constructor(keyPair, type) {
+        /** @type {Nimiq.KeyPair} */
+        this._keyPair = keyPair;
+        /** @type {Nimiq.Address} */
+        this.address = this._keyPair.publicKey.toAddress();
+        /** @type {string} */
+        this.userFriendlyAddress = this.address.toUserFriendlyAddress();
+        this.type = type;
+    }
+
+    /**
+     * Sign a generic message
+     * @param {string} message - A utf-8 string
+     * @returns {object}
+     */
+    signMessage(message) {
+        message = 'nimiq_msg_' + message;
```

Comment:
To prevent the user from accidentally signing arbitrary data (like transactions), we currently only allow strings and prepend a message identifier. Ethereum and Bitcoin do the same (in proper implementations). These prefixes are hidden from the user usually when displaying the signed data.